handlers/modals.go

```diff
@@ -23,45 +25,49 @@ func UsersModalAddValidator(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// const VALIDATOR_EVENTS = ['validator_attestation_missed', 'validator_proposal_missed', 'validator_proposal_submitted', 'validator_got_slashed', 'validator_synccommittee_soon']
-	// const MONITORING_EVENTS = ['monitoring_machine_offline', 'monitoring_hdd_almostfull', 'monitoring_cpu_load']
-
 	validatorForm := r.FormValue("validator")
 
 	validators := []string{}
-
+	invalidValidators := []string{}
 	for _, userInput := range strings.Split(validatorForm, ",") {
 
 		if utils.IsValidEnsDomain(userInput) || utils.IsEth1Address(userInput) {
 			searchResult, err := FindValidatorIndicesByEth1Address(userInput)
-			if err == nil {
-				for _, res := range searchResult {
-					for _, index := range res.ValidatorIndices {
-						validators = append(validators, fmt.Sprintf("%v", index))
-					}
+			if err != nil {
+				invalidValidators = append(invalidValidators, userInput)
+				continue
+			}
+			for _, res := range searchResult {
+				for _, index := range res.ValidatorIndices {
+					validators = append(validators, fmt.Sprintf("%v", index))
 				}
 			}
-
-		} else {
+		} else if _, err := strconv.Atoi(userInput); err == nil {
 			validators = append(validators, userInput)
+		} else {
+			invalidValidators = append(invalidValidators, userInput)
 		}
 	}
+	if len(invalidValidators) > 0 {
+		utils.SetFlash(w, r, authSessionName, fmt.Sprintf("Error: Invalid validator(s) %v, please try again.", strings.Join(invalidValidators, ", ")))
+		http.Redirect(w, r, "/user/notifications", http.StatusSeeOther)
+		return
+	}
 
 	for _, val := range validators {
 		pubkey, _, err := GetValidatorIndexFrom(val)
 		if err != nil {
-			utils.LogError(err, "error parsing form", 0)
-			utils.SetFlash(w, r, authSessionName, "Error: Something went wrong adding your validator to the watchlist, please try again in a bit.")
+			if err == sql.ErrNoRows {
+				logger.Warnf("Could not find validator %v when trying to add to watchlist", val)
+				utils.SetFlash(w, r, authSessionName, fmt.Sprintf("Error: Could not find validator %v, please try again.", val))
```

Comment:
Adding the sentence "No validators added to the watchlist, please try again." improves this, but still, only 1 invalid validator is mentioned in the first part of the error message: "Could not find validator 999997."
Why not let the user know about every validator that wasn't found?

Furthermore, while pondering on this I found another problem: Try to add "999,999999,888" on Prater. This is an existing validator followed by a validator that does not exist followed by another existing validator.
The error message in this case states "Could not find validator 999999. No validators added to the watchlist, please try again."
Validator 999 has been added though.

The root of this issue is the way the loop here works: https://github.com/gobitfly/eth2-beaconchain-explorer/blob/76e74910164abad2e678b45b693ce2ddaa81b913/handlers/modals.go#L62

I guess the easiest way to fix this would be to split this up in two loops: First loop over all validators and try to get their pubkey. If a single validator fails, do nothing and show an error message containing all missing validators (with a smart validator/validators text maybe?).
If all validator pubkeys are retrieved successfully, loop over all of them again and handle the subscription,

If something goes wrong there, we ignore the "999,999999,888" issue just mentioned for the sake of simplicity. This should be fine as errors here most likely are caused by issues with the db. In that case no validator gets added most likely anyways.

Thoughts?

Reply:
splitted into two loops, ptal